pkg/storaged/client.js

```diff
@@ -153,13 +153,42 @@ define([
      */
 
     function update_indices() {
-        var path, block, dev, mdraid, vgroup, pvol, lvol, part;
+        var path, block, dev, mdraid, vgroup, pvol, lvol, part, i;
 
+        client.drives_multipath_blocks = { };
         client.drives_block = { };
+        for (path in client.drives) {
+            client.drives_multipath_blocks[path] = [ ];
+        }
         for (path in client.blocks) {
             block = client.blocks[path];
-            if (block.Drive != "/")
-                client.drives_block[block.Drive] = block;
+            if (!client.blocks_part[path] && client.drives_multipath_blocks[block.Drive] !== undefined)
+                client.drives_multipath_blocks[block.Drive].push(block);
+        }
+        for (path in client.drives_multipath_blocks) {
+            var all_blocks = client.drives_multipath_blocks[path];
+            var mpath_members = [ ];
+            var non_mpath_members = [ ];
+
+            all_blocks.sort(function (a, b) { return a.DeviceNumber - b.DeviceNumber; });
```

Comment:
Yes, agreed.  This is a common pattern in the storage code.  I'll factor it out and document it.